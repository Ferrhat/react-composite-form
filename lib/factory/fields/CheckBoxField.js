import React, {Component} from 'react';
import Switch from 'react-ios-switch';

class CheckBoxField extends Component {
    render() {
        const { name, label, value, valueUpdateHandler, changeHandler, disabled, options } = this.props;

        const onChange = (event) => {
            valueUpdateHandler(name, event);
            changeHandler(event);
        }

        return (
            <Switch
                checked={ value }
                onChange={ onChange }
                disabled={disabled}
                onColor={'#00C9FF'}
            />
        );
    }
}

export default CheckBoxField;
