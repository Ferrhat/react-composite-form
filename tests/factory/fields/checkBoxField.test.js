import React from 'react';
import {shallow, mount} from 'enzyme';
import CheckBoxField from "../../../lib/factory/fields/CheckBoxField";


describe('CheckBoxField', () => {
    const mockValueUpdateHandler = jest.fn();
    const mockChangeHandler = jest.fn();
    const checkBoxField = shallow(<CheckBoxField valueUpdateHandler={mockValueUpdateHandler} changeHandler={mockChangeHandler} name={'testName'} />);

    it('renders properly', () => {
        expect(checkBoxField).toMatchSnapshot();
    });

    it('contains a connected Switch component', () => {
        expect(checkBoxField.find('Switch').length).toEqual(1);
    });

    it('should trigger a change event', () => {
        checkBoxField.find('Switch').simulate('change', true);
        expect(mockValueUpdateHandler).toBeCalledWith('testName', true);
        expect(mockChangeHandler).toBeCalledWith(true);
    });

});
