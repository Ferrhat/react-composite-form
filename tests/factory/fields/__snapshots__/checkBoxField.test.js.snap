// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxField renders properly 1`] = `
<Switch
  handleColor="white"
  offColor="white"
  onChange={[Function]}
  onColor="#00C9FF"
/>
`;
